### ISSUE TITLE
listPerson cleanup

### ISSUE THREAD

**Issue by matyaskopp** [ufal/ParlaMint-UA#45]
in [ParlaMint-UA-listPerson.xml](https://github.com/ufal/ParlaMint-UA/blob/d58289b7f42621bd5a052ab8127dd17fece12911/ParlaMint-UA.TEI/ParlaMint-UA-listPerson.xml) there are currently 
```bash
cat /opt/ParlaMint-UA/ParlaMint-GIT/DataRelease/ParlaMint-UA/ParlaMint-UA-listPerson.xml |grep -o 'person [^>]*'|wc -l
```
`2192` persons and in the whole corpus, there are 
```bash
cat utterance-who-ana/20230212T215214/who-ana-cnt.tsv |tail -n +4|cut -f 1|sort|uniq|wc -l
```
`1477` different speakers.

```bash
java -cp /usr/share/java/saxon.jar net.sf.saxon.Query -xi:off \!method=adaptive -qs:'//*[local-name()="person"][./*[local-name()="affiliation" and contains(" #ВРУ #GOV.UA ", @ref) and (not(@to) or @to >= "2012-12-12")]]/@xml:id' -s:/opt/ParlaMint-UA/ParlaMint-GIT/DataRelease/ParlaMint-UA/ParlaMint-UA-listPerson.xml|wc -l
```
`1599` persons are affiliated with parliament or government in period starting in 2012-12-12.

So there should be
```bash
(cat utterance-who-ana/20230212T215214/who-ana-cnt.tsv |tail -n +4|cut -f 1 ; java -cp /usr/share/java/saxon.jar net.sf.saxon.Query -xi:off \!method=adaptive -qs:'//*[local-name()="person"][./*[local-name()="affiliation" and contains(" #ВРУ #GOV.UA ", @ref) and (not(@to) or @to >= "2012-12-12")]]/@xml:id' -s:/opt/ParlaMint-UA/ParlaMint-GIT/DataRelease/ParlaMint-UA/ParlaMint-UA-listPerson.xml|sed 's/^[^"]*"//;s/"//')|sort|uniq|wc -l
```
`1809` persons in `listPerson` when all guests and government/parliament members are included

The current listPerson also includes MPs from previous periods (they can be guests, so it is necessary for processing linking persons with speakers). 

@AnnaParla, 
can we defend having these MPs outside a focussed period in data?
The number (`2192 -1809 = 383` ) is smaller than I expected (persons from 4-6 convocations that are not in 7-9 convocations) - probably only small political changes in Ukraine in these periods...

**Comment by AnnaParla** [comment]
<@AnnaParla,
can we defend having these MPs outside a focussed period in data?
The number (2192 -1809 = 383 ) is smaller than I expected (persons from 4-6 convocations that are not in 7-9 convocations) - probably only small political changes in Ukraine in these periods...>

Yes, the percentage of new faces in the Rada was lower back then, with term 6 being all time low. Here are the numbers:
![image](https://user-images.githubusercontent.com/115141649/224265152-0fb7549d-54e7-4af0-86a7-8c079c05a49a.png)
But it seems that the number you have is mathematically even lower. Do you have all the data for convocations 4-6?

**Comment by matyaskopp** [comment]
> But it seems that the number you have is mathematically even lower. Do you have all the data for convocations 4-6?

Yes, MPs' info from convocations 4-6 is completely downloaded. We don't have information only about the government from this period.
I wanted to have older data, but they are published in a different format.

**Comment by AnnaParla** [comment]
Ok, and what is the timeframe for <2192 persons and in the whole corpus>?

**Comment by matyaskopp** [comment]
> Ok, and what is the timeframe for <2192 persons>?

446 affiliations with Rada starts on 2002-05-14. So the timespan starts on 2002-05-14 and ends on 2023-02-24 (last text).
